src/TheThingsNetwork.cpp

```diff
@@ -396,19 +402,47 @@ void TheThingsNetwork::autoBaud()
   while (attempts-- && length == 0)
   {
     delay(100);
+    // Send break + Autobaud 
     modemStream->write((byte)0x00);
     modemStream->write(0x55);
     modemStream->write(SEND_MSG);
-    sendCommand(SYS_TABLE, 0, true, false);
-    sendCommand(SYS_TABLE, SYS_GET, true, false);
-    sendCommand(SYS_TABLE, SYS_GET_VER, false, false);
-    modemStream->write(SEND_MSG);
-    length = modemStream->readBytesUntil('\n', buffer, sizeof(buffer));
+    // check we can talk
+    length = checkModuleAvailable();
+    
+    // We succeded talking to the module ?
+    baudDetermined = length > 0 ;
   }
   delay(100);
   clearReadBuffer();
   modemStream->setTimeout(10000);
-  baudDetermined = true;
+}
+
+bool TheThingsNetwork::isSleeping()
+{  
+  return baudDetermined ? false : true ;
+}
+
+void TheThingsNetwork::wake()
+{  
+  // If sleeping
+  if (!baudDetermined) 
+  {
+    // Send a 0 at lower speed to be sure always received
+    // as a character a 57600 baud rate
+    modemStream->flush();
+    modemStream->begin(2400);
+    modemStream->write((uint8_t) 0x00);
+    modemStream->flush();
+    delay(20);
+    // set baudrate back to normal and send autobaud
+    modemStream->begin(57600);
```

Comment:
Again, `Stream` does not have `begin()`, right? If so, where is it defined?

I.e. this code is still breaking when `SerialType` cannot be determined and falls back to `Stream`.

Reply:
Correct tested with a defined symbol when a hardware serial is used 
take care, now this will compile for SparkFun Micro but won't work since the begin will not be executed. For this to work users need to select new TTN boards in IDE ;-)

Reply:
Why would that not work? In those cases, `HARDWARE_UART` gets defined on https://github.com/TheThingsNetwork/arduino-device-lib/pull/231/files/bab08ab865752ff5208c71f6f9dae807e6be8fba..a164a60fd13751bf79ae74f4b50bb9afcabd29ed#diff-49a17fc31f5e939a36dc3a3e67227c1bR30